pyTigerGraph/pyTigerGraph.py

```diff
@@ -960,54 +987,123 @@ def runInterpretedQuery(self, queryText, params=None):
+                       variables or accumulators, or plain text printed) be captured as well.
 
-        NOTE: The primary ID of a vertex instance is NOT an attribute, thus cannot be used in above arguments.
-              Use `getVerticesById` if you need to retrieve by vertex ID.
-
-        Endpoint:      GET /graph/{graph_name}/vertices
-        Documentation: https://docs.tigergraph.com/dev/restpp-api/built-in-endpoints#get-graph-graph_name-vertices
+        Returns: A dictionary with two (or three) keys: "Vertices", "Edges" and optionally "Output". First two refer to another dictionary
+            containing keys for each vertex and edge types found, and the instances of those vertex and edge types. "Output" is a list of
+            dictionaries containing the key/value pairs of any other output.
         """
-        data = self.getVertices(vertexType, select=select, where=where, limit=limit, sort=sort, timeout=timeout)
-        df = pd.DataFrame(data)
-        df = pd.concat([df["v_id"], pd.DataFrame(df["attributes"].tolist())], axis=1)
-        return df
+        vs = {}
+        es = {}
+        ou = []
+        # Outermost data type is a list
+        i = 0
+        for o1 in output:
+            # Next level data type is dictionary
+            for o2 in o1:
+                o3 = o1[o2]
+                if not isinstance(o3, list):
+                    if not graphOnly:  # Vertices and edges are coming in lists; but complex data types too, so more check are needed later
+                        ou.append({o2: o3})
+                else:
+                    ox = o3[0]
+                    if not isinstance(ox, dict):
+                        ou.append({o2: o3})
+                    else:
+                        if "v_type" in ox:
+                            for o4 in o3:
+                                vt = o4["v_type"]
+                                if vt not in vs:
+                                    vs[vt] = []
+                                vs[vt].append(o4)
+                        elif "e_type" in ox:
+                            for o4 in o3:
+                                et = o4["e_type"]
+                                if et not in es:
+                                    es[et] = []
+                                es[et].append(o4)
+                        elif not graphOnly:
+                            ou.append({o3: ox})
+            i += 1
+        ret = {"Vertices": vs, "Edges": es}
+        if not graphOnly:
+            ret["Output"] = ou
+        return ret
 
-    def getVertexDataframeByID(self, vertexType, vertexIds):
-        """Retrieves vertices of the given vertex type, identified by their ID.
+    # Pandas DataFrame support =================================================
 
-        Arguments
-        - `vertexType`: Type of vertex desired
-        - `vertexIds`: A single vertex ID or a list of vertex IDs.
+    def vertexSetToDataFrame(self, vertexSet, withId=True, withType=False):
```

Comment:
This seems like the standard way things from out from queries. Please explain that in the docstring.

Reply:
That is not certainly true. A query might return a vertex set, or might not.
Provided more details/explanation for this function.

Reply:
This docstring works for me. Thanks!